contracts/integration/MoC/MocIntegration.sol

```diff
@@ -0,0 +1,107 @@
+     * @param _doc DoC contract address
+     * @param _dllr DLLR contract address
+     * @param _massetManager MassetManager contract address
+     */
+    constructor(address _moc, address _doc, address _dllr, address _massetManager) {
+        require(
+            _moc != address(0) &&
+                _doc != address(0) &&
+                _dllr != address(0) &&
+                _massetManager != address(0),
+            "MocIntegration:: no null addresses allowed"
+        );
+        moc = IMocMintRedeemDoc(_moc);
+        doc = IERC20(_doc);
+        dllr = IDLLR(_dllr);
+        massetManager = IMassetManager(_massetManager);
+    }
+
+    ///@dev the contract requires receiving funds temporarily before transferring them to users
+    receive() external payable {}
+
+    /**
+     * @notice how getDocFromDllrAndRedeemRBTC function works:
+     * -------------------------------------------------------------------------------------------
+     * |               Mynt                         |                Money On Chain              |
+     * -------------------------------------------------------------------------------------------
+     * | get DLLR (EIP-2612) -> convert DLLR to DoC | -> get RBTC from DoC -> send RBTC to user  |
+     * -------------------------------------------------------------------------------------------
+     *
+     * @param _dllrAmount The amount of the DLLR (mAsset) that will be burned in exchange for _toToken
+     * @param _permitParams EIP-2612 permit params:
+     *        _deadline Expiration time of the signature.
+     *        _v Last 1 byte of ECDSA signature.
+     *        _r First 32 bytes of ECDSA signature.
+     *        _s 32 bytes after _r in ECDSA signature.
+     */
+    function getDocFromDllrAndRedeemRBTC(
+        uint256 _dllrAmount,
+        PermitParams calldata _permitParams
+    ) external {
+        // transfer _dllrAmount to this contract by permit (EIP-2612)
+        address thisAddress = address(this);
+        uint256 dllrBalanceBefore = dllr.balanceOf(thisAddress);
+        dllr.transferWithPermit(
+            msg.sender,
+            thisAddress,
+            _dllrAmount,
+            _permitParams.deadline,
+            _permitParams.v,
+            _permitParams.r,
+            _permitParams.s
+        );
+        require(
+            dllr.balanceOf(thisAddress) == dllrBalanceBefore + _dllrAmount,
+            "MocIntegration:: DLLR transfer with permit received wrong amount"
+        );
+
+        // redeem DoC from DLLR
+        require(
+            massetManager.redeemTo(address(doc), _dllrAmount, thisAddress) == _dllrAmount,
+            "MocIntegration:: redeemed incorrect DoC amount"
+        );
+
+        // redeem RBTC from DoC using Money On Chain and send to the user
+        uint256 rbtcBalanceBefore = thisAddress.balance;
+        // TODO: clarify the vendor address param
+        moc.redeemFreeDocVendors(_dllrAmount, payable(address(0)));
```

Comment:
jamie is also using the 0 address. currently, it's the right thing to do, but i guess it could change in the future. i think it would be more future-proof if we make the vendor address settable.

Reply:
added mocVendorAccount storage variable & setter